statsd/statsd.go

```diff
@@ -101,6 +101,13 @@ const (
 	serviceCheck
 )
 
+type SendingMode int
+
+const (
+	BlockOnSend SendingMode = iota
```

Comment:
Maybe too much of a nitpick but I find those names confusing:
- "Block" can be interpreted as blocking on network (neither mode will block on network)
- Both modes will drop metrics on send if queues get full
- Chan are not lock-free, the contention will just be lower

Reply:
I struggled a bit on the naming. Do you have any proposition ? This is the best I could come up with.

Reply:
IMO we can use Mutex & (Non-Blocking) Channel in the naming. Like you said this is a very specialized thing that we don't expect regular people to use so I don't think we need to hide the technical details in the names